CARDS-1731: Refactor the PROMs dashboard to display appointments and surveys by default

marta- opened the pull request:
**What changed:**
* Proms dashboard widgets for appointments and surveys no longer need to be specified via extensions, they are displayed by default
* An extension point is still available for adding more widgets to the dashboards
* The url of the clinic dashboard is now specified using the node name for the clinic configuration

**Testing:**
* Everything on the proms dashboard should work as before. The correct display of existing appointments and Survey responses (in the right time tabs) must be tested.

**Implications:**
* The code in PR #983 for configuring the dashboard would have to be modified as follows:
  * Proms dashboard extensions no longer need to be generated
  * Proms dashboard extension point for a specific clinic now has a different node name (`DashboardViews<hash of clinic id>`)
  * Sidebar label link now points to a different URL (`/Dashboard/<hash of clinic id>`)

marta- committed:
CARDS-1731: Refactor the PROMs dashboard to display appointments and surveys by default

sashaandjic reviewed (approved):
tested
![image](https://user-images.githubusercontent.com/23243244/164580178-7ce3fcdd-daf4-4c77-9169-7251da0cfc54.png)

marta- committed:
CARDS-1731: Refactor the PROMs dashboard to display appointments and surveys by default
Fixing rebase bug: `surveyID` got renamed to `survey` in the clinics configuration

sdumitriu commented on proms-resources/frontend/src/main/frontend/src/proms/PromsDashboard.jsx:
```diff
@@ -159,6 +181,18 @@ function PromsDashboard(props) {
+        </Grid>
+        {/* Survey views */}
+        { surveys?.map((s, index) => (
+            <Grid item lg={12} xl={6} key={`view-survey-${index}`} className={classes.dashboardEntry}>
```
Bug:
- create a new clinic using the `Mood` qset
- open the Mood dashboard -> 1+3 widgets displayed, the correct ones (PHQ, GAD, AUDIT)
- switch to the PMCC dashboard -> 1+5 widgets displayed, but the wrong ones; the first 3 ones are the same as before, then AUDIT again, the SC
- reload the page -> 1+5 widgets are displayed, the correct ones
- switch to the Mood dashboard -> 1+3 widgets displayed, but the wrong ones, the first 3 from Cardio

I think that using a different, more meaningful `key` here may help fix the problem. Perhaps `clinicId + surveyId`?

sdumitriu committed:
CARDS-1731: Refactor the PROMs dashboard to display appointments and surveys by default
Bugfix: switching between clinics with different questionnaire sets ends up displaying the wrong questionnaire tables

sdumitriu committed:
CARDS-1731: Refactor the PROMs dashboard to display appointments and surveys by default
Bugfix: switching between clinic dashboards caches the livetable column definition, leading to wrong columns being displayed

OrdiNeu committed:
CARDS-1731: Refactor the PROMs dashboard to display appointments and surveys by default

Remove no longer needed code creating explicit extensions for each questionnaire
